02_rand_dataset.ipynb

```diff
@@ -32,57 +23,401 @@
xcXGjVqRE5Ojj5V\nB2D8+PHMnj2bypUrG6o7r9Xjx48ZPnw4169fB55/L9LT09m6dStfffUV69evJzIyklKlShEZGYlG\no+Hdd9/l8uXLxMfH69+nQ4cOuLm5vVAGT7x+f1R2a+/eveTm5vLBBx+gUqm4du0aFy5cIDo6mvPn\nz5Obm8vq1atZsGABycnJTJw4EZVK9VYukjK0S5cukZCQwPfff0+XLl148OABkZGRnD59GicnJ1au\nXElISAh2dnasXLmSffv2kZ6eTlRUFK6urnh7exu6CyWOWq1+ZTna/fv3c+/ePaKjo5k/fz4rVqzg\nxIkT7N69G29vbwICAjhy5AinT5/Wv2b06NFcuHCBK1euvPF+/F2SziFeK910nFKp5PHjx9y9excj\nIyNMTU1JTU0lLi6ORo0aUalSJQCqVq3KwYMHuXDhAi1btkSlUhEeHs727duJiYmhX79+tGzZEicn\nJ1l1bWB5eXkkJSXh6OioX2ilVCrJyclhypQpLF26lMTERFxdXenduzcKhYLZs2djYWFBrVq1SEpK\nIi8vD29vbxwcHLhz5w5RUVG0aNECKysrlEol5cqVM3Q3Xxtra2suXbpEZGQkMTEx2Nvb88svv7B3\n717CwsLIyclh7ty5nD17ltTUVMqXL0+LFi24cOECV69excXFRT+t2alTJ0nn+Ifoziu6c8vWrVv5\n4YcfyM7OpnLlyhgZGZGXl4enpydLlizhypUr9OnTh6dPn5KcnEyrVq1ISkpi165d7Ny5EzMzM+bO\nnYujo6OBe/Z2+eWXXwgODiY+Pp4RI0bQq1cvWrZsiZ2dHd988w21a9fm0qVLeHp60rp1a5KTk1m+\nfDl79uzh6NGjjBs3jjp16hi6GyXC0aNHmT17Nm3btkWj0ehrpcOvN6Hbtm0jJyeHnj17cuzYMcLD\nw0lPTyctLY3AwEDq1KnDrl27yMvLw8vLCzMzM+zt7enfv/9bdR2QIFq8Vrov0Lx58/i///s/Tp06\nxbp167Czs8Pf35/IyEiUSiV169bVL9axt7dnyZIl1KxZky5duhAYGIinpyfTp0/Hy8vrpfcWb15e\nXh6LFi3iyy+/ZNCgQahUKrRaLT/99BN9+/ZFqVTSpk0bDh06RHJyMr6+vvj7+2NhYcHBgweJj4+n\nSpUqlC5dGg8PD30FA1dXV+rVq/fCSfhtphuZ1/33u+++04+ijR07FoDatWtja2tLREQEz549Y/ny\n5SQmJnLo0CFatWpFtWrViIiIwNbWFh8fH5RKpawH+If8duMOjUbDnDlziIiIIDs7m4iICGrVqoW/\nvz+BgYH8+OOPHDt2jN69e9O+fXv27NnDoUOHcHV1pXPnznTo0IE2bdrw7rvvlqhSja+LmZkZtra2\nREdH89577+Hs7IyRkRGVK1cmMTGRK1euUKdOHQ4ePEj37t1p3rw5wcHB1KtXj8mTJ5fYOvNv0pUr\nVyhdujR37txhzpw5+Pj4ULFiRY4ePUp4eDhnzpzBysqKsmXLYm1tTUBAALdu3SIyMpKgoCCmT5/O\nwoULMTY2pkWLFjx9+pQdO3ZQs2ZNXFxcAN66c50E0eJ/8qrR4djYWL799lsmT57M+++/j1KpZP78\n+fqAaeHChfj4+ODq6go836zg8OHD3LlzhzZt2uDm5ka1atVQKpWvLC0l3jyVSoWxsTFnzpzh4cOH\n+Pv7o9Fo+Oabb3B3d2fu3Ln4+vpy//59YmJisLS0xNfXFx8fH5ydndm7dy/R0dE4ODjQvHlzAFxc\nXPD29n6rA+j4+Hisra0xNzd/YSV6Tk6O/mJgaWnJxYsX6dOnD/b29ri5ubFlyxYOHz7Mv/71L9zc\n3Lh69So7d+7k8ePHdOzYkRo1atC2bdtiu8DS0HRlt3SLPNeuXauvtDFr1iz69etHVlYWq1evJiQk\nBIVCwciRI2nbti2dO3emoKCAnTt3olQqOXDgACEhIZQqVQorKytDd+2tVr16dXbu3ImpqSlBQUGo\n1Wr9TIDu5jw+Ph5jY2O8vLywtbXVB19yrfhnXbx4kREjRuiLADx58oSVK1fi7+/PmDFjqFq1Kpcu\nXSI2NpYHDx7QsWNHrKys+Oyzz3B2dqZ79+44OjqyY8cOLl68iFKp5N1338Xd3f2tnmWTT5z4W7Ra\nLQUFBa8cHd6yZQsNGzakadOm3L17l507d2Jubo6FhQXBwcEEBQUxb948Jk6cyPbt24mNjWX27Nn8\n3//930sXId1otTAcXbmpunXr0rFjR7777jt+/vlnVCoVKSkpODk5oVarWbp0KdHR0bi5uXH06FEu\nXLigf92cOXNo0KABP/30E48fP37h/d/GclRpaWm0adOG8PBw/e9HoVCwb98+evXqxbhx49i2bRsd\nO3Zk+vTpqNVqFi9eDDwPsM+dO0eDBg3w8/MD4Nq1azRs2BAbGxuys7OpX78+pqamButfcacLtg4e\nPEjXrl2JjIxk7NixnDlzBnd3d8zMzBg5ciTZ2dmsWLFCn+N8+PBhkpOTmTp1KhkZGcycOZM9e/bI\nsXpNTE1N+eCDD/jmm2+4f/++/gY7KSmJ0qVL06pVK7p27Uq9evVeeq1cK/4ZO3bs4IcffsDLy4st\nW7ZQo0YNjI2NGTBgAPn5+bz//vt0796dzz//nOXLl9O7d29WrlzJhQsXyMzMJD09neDgYMqXL8+5\nc+ews7PDy8sLU1NTlEolDRo0MHQX/ycyEi0K7bd5z5mZmaxfv54nT55gbm6OpaUl58+fJycnh/37\n9zNt2jQaNmzIzJkzuX//PmlpafTu3RtjY2POnTvHkSNHaNq0Kf369Xur8qCKK41Gw86dO/Ul1uD5\nKLRarWbRokU8fvyYc+fOkZ2dTePGjfH09MTX15ft27dz4cIFxo4dS6NGjdiwYQMmJiYEBAQAYGtr\ni729Pbdu3SIoKOiFm6W
```

Comment:
One space is still missing (location_entity_full_`df,annot`=True --> location_entity_full_`df, annot`=True):
```
hmap = sns.heatmap(location_entity_full_df, annot=True, fmt='d', cmap='YlGnBu', cbar=False)
```